cloudbuild.yaml

```diff
@@ -1,17 +1,42 @@
-# This file is used to build and deploy the app into a GKE cluster using Google
-# Cloud Build.
+# Cloudbuild.yaml to deploy to staging
 #
-# Requirements:
-# - Give the Google Cloud Build service account "Kubernetes Engine Developer"
-#   IAM role on the GCP project.
-# - Set up Google Cloud Build trigger on Cloud Console.
+# PREREQUISITES:
+# - Cloud Build service account must have roles: "Kubernetes Engine Developer", "Storage Object Creator", \
+#   "Storage Object"
 
 steps:
-- name: gcr.io/k8s-skaffold/skaffold:v0.16.0
-  args: ['skaffold', 'run', '-f=skaffold.yaml']
+
+# authenticate to GKE cluster
+- id: 'Get GKE credentials'
+  name: gcr.io/cloud-builders/gcloud
+  args: ['container','clusters','get-credentials','demo-app-staging']
   env:
   - 'CLOUDSDK_COMPUTE_ZONE=us-central1-b'
+
+# make latest version of the Skaffold community builder available in this project
+- id: 'clone community builders repo'
+  name: gcr.io/cloud-builders/git
+  args: ['clone','https://github.com/GoogleCloudPlatform/cloud-builders-community']
+
+- id: 'build Skaffold builder'
```

Comment:
why not stick with gcr.io/k8s-skaffold/skaffold image?

Reply:
I ran into a problem with that... because it's not intended for use as a Cloud Build builder, it failed when used inside GCB (I'm afraid I don't recall what the difference was, or how it failed -- that was pre-Thanksgiving).

Reply:
It might be because I keep changing the skaffold.yaml spec version and updating skaffold version in .travis.yml.

I strongly recommend using the off-the-shelf image. it saves us 2 steps == better sample readability.